core/src/display_object/edit_text.rs

```diff
@@ -1183,15 +1183,141 @@ impl<'gc> EditText<'gc> {
         None
     }
 
-    pub fn text_input(self, character: char, context: &mut UpdateContext<'_, 'gc>) {
-        if self.0.read().flags.contains(EditTextFlag::READ_ONLY) {
+    fn available_chars(self) -> usize {
+        let read = self.0.read();
+        let max_chars = read.max_chars;
+        if max_chars == 0 {
+            usize::MAX
+        } else {
+            let text_len = read.text_spans.text().len();
+            let selection_len = if let Some(selection) = self.selection() {
+                selection.end() - selection.start()
+            } else {
+                0
+            };
+            max_chars.max(0) as usize - (text_len - selection_len)
```

Comment:
I am not certain that I understand this part of the code. But, can this overflow? Like, if `max_chars.max(0)` can yield `1`, `text_len` (can the text be set by a script, thus bypassing the `max_chars`limit? https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/text/TextField.html#maxChars . Though I might be misunderstanding the code) can yield `10`, and `selection_len` can yield 0, that would give: `1 as usize - (10 - 0)` which would evaluate to something negative despite it being unsigned.

Reply:
Oh good point, I didn't think of that